### PR TITLE
RFC: osd_directory for docker

### PR DESCRIPTION
**Pull request by ibotty**
I think it might make sense to consolidate the docker and non-docker scenarios, so I don't think it makes sense to merge as is. Please comment on how to proceed.

### CODE REVIEW

File: roles/ceph-osd/tasks/check_mandatory_vars.yml [hunk 1]
```diff
@@ -1,15 +1,4 @@
 ---
-- name: make sure public_network configured
-  fail:
-    msg: "public_network must be configured. Ceph public network"
-  when: public_network == '0.0.0.0/0'
```
Comment (leseb) [inline roles/ceph-osd/tasks/check_mandatory_vars.yml]: why do you remove these lines?

File: roles/ceph-osd/tasks/check_mandatory_vars.yml [hunk 2]
```diff
@@ -1,15 +1,4 @@
...
-
-- name: make sure cluster_network configured
-  fail:
-    msg: "cluster_network must be configured. Ceph replication network"
-  when:
-    - cluster_network == '0.0.0.0/0'
-    - not osd_containerized_deployment
```
Comment (leseb) [inline roles/ceph-osd/tasks/check_mandatory_vars.yml]: ditto

File: roles/ceph-osd/tasks/check_mandatory_vars.yml [hunk 3]
```diff
@@ -62,7 +51,7 @@
   when:
     - osd_group_name is defined
     - osd_group_name in group_names
-    - (journal_collocation or osd_containerized_deployment)
+    - (journal_collocation or (osd_containerized_deployment and not osd_directory))
```
Comment (leseb) [inline roles/ceph-osd/tasks/check_mandatory_vars.yml]: why not moving `not osd_directory` bellow?

File: roles/ceph-osd/tasks/docker/osd_directory.yml [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    -v "{{ item }}:{{ item }}"
+    --entrypoint /bin/sh
+    "{{ ceph_docker_registry }}/{{ ceph_osd_docker_username }}/{{ceph_osd_docker_imagename }}:{{ ceph_osd_docker_image_tag }}"
+    -c 'ceph-disk prepare --cluster "{{ cluster }}" "{{ item }}"'
+  args:
+    creates: "{{ item }}/superblock"
+  with_items: "{{ osd_directories }}"
+  when: osd_directory
+
+- name: activate OSD(s)
```
Comment (leseb) [inline roles/ceph-osd/tasks/docker/osd_directory.yml]: Why do you need to activate this? Preparing and running the container through systemd should be sufficient right?

File: roles/ceph-osd/tasks/docker/start_docker_osd.yml [hunk 5]
```diff
@@ -67,6 +69,13 @@
     - ceph_osd_docker_prepare_env is defined
     - osd_containerized_deployment_with_kv
 
+- include: osd_directory.yml
+  when: osd_directory
+
+- name: XXX
```
Comment (leseb) [inline roles/ceph-osd/tasks/docker/start_docker_osd.yml]: I know this is not complete but what do you want to debug here?

**Comment by ibotty** [comment]
Oh, I accidentally included some local changes (for testing) when rebasing. Sorry. The intent is clear I hope though.

**Comment by leseb** [comment]
I really hope to merge https://github.com/ceph/ceph-ansible/pull/1287 today, so don't bother yet :)

**Comment by leseb** [comment]
Closing this due to non-activity, feel free to re-open.

**Pull request closed without merging**